### PR TITLE
feat: add emulator provider

### PR DESCRIPTION
**Pull request by hadelive**

### CODE REVIEW

File: packages/lucid/src/lucid-evolution/LucidEvolution.ts [hunk 1]
```diff
@@ -81,6 +82,14 @@ export const Lucid = async (
     txbuilderconfig: TxConfig.makeTxConfig(protocolParam, costModels),
     protocolParameters: protocolParam,
   };
+  if ("slot" in config.provider) {
```
Comment (hadelive) [inline packages/lucid/src/lucid-evolution/LucidEvolution.ts:85]: Unable to use `if (config.provider instanceof Emulator)` , so I use the property check instead.

**Comment by solidsnakedev** [comment]
@hadelive we should move at the transaction test to lucid package, and leave the provider test only. please have a look at the kupmios provider test

**Comment by solidsnakedev** [comment]
also ideally we should add a dependency injection like this https://github.com/Anastasia-Labs/lucid-evolution/blob/301b07bba5a92b8dbdc7cee914b1aa9169b66016/packages/lucid/test/specs/services.ts#L17-L34
so we can have both onchain test and emulator test

File: packages/provider/test/service.ts
Comment (solidsnakedev) [inline packages/provider/test/service.ts:1]: could you move this to `packages/lucid/test` ? I'm ok if you make an emulator folder with all the test transactions

File: packages/provider/test/emulator.test.ts
Comment (solidsnakedev) [inline packages/provider/test/emulator.test.ts:1]: could you move this to packages/lucid/test ? I'm ok if you make an emulator folder with all the test transactions

**Review by solidsnakedev — APPROVED** [review]

**Pull request merged**